src/vs/base/browser/indexedDB.ts

```diff
@@ -117,6 +118,60 @@ export class IndexedDB {
 		}
 		const transaction = this.database.transaction(store, transactionMode);
 		this.pendingTransactions.push(transaction);
+		// MEMBRANE: List of keys we save using our `/settings` API endpoint
+		// indexeddb name
+		const dbName = this.name;
+		const MEMBRANE_KEYS = [
+			// 'memento/webviewView.membrane.logs',
+			// 'memento/webviewView.membrane.navigator',
+			// 'memento/webviewView.membrane.packages',
+			'/User/settings.json'
+		];
+		// MEMBRANE: Proxy
+		const storeProxy = new Proxy(transaction.objectStore(store), {
+			get(target: any, prop: any): any {
+				if (typeof target[prop] === 'function') {
+					return (...args: any[]) => {
```

Comment:
The idea we discussed:

 Instead of calling the API here, just called all requests in an array: `Array<prop, key, IDBRequest>`. And then execute the actual API calls in the `oncomplate` callback.